src/ad-cid.js

```diff
@@ -29,13 +29,25 @@ export function getAdCid(adElement) {
   if (!config || !config.clientIdScope) {
     return Promise.resolve();
   }
-  const cidPromise = cidForDocOrNull(adElement.getAmpDoc()).then(cidService => {
+  return getAdCidHelper(adElement.getAmpDoc(), config.clientIdScope,
+                        config.clientIdCookieName);
+}
+
+/**
+ * @param {!./service/ampdoc-impl.AmpDoc} ampDoc
+ * @param {!string} clientIdScope
+ * @param {string=} opt_clientIdCookieName
+ * @return {!Promise<string|undefined>} A promise for a CID or undefined.
+ */
+export function getAdCidHelper(
```

Comment:
I would expect a method named "getXXHelper" returns a util object for XX.

`getOrCreateAdCid` is actually a good name, it returns undefined only when there's an error (service does not exist, or timed out).